doc/network-ng.md

```diff
@@ -40,29 +40,73 @@ other classes.
 
 ## Backend Support
 
-In order to support a new backend, we need to implement a configuration readers and writers. The
-{Y2Network::Sysconfig} module implements support to deal with `sysconfig` files.
+As mentioned above, Y2Network is designed to support different backends. It is expected to implement
+a reader and a writer for each backend (except for AutoYaST, which is an special case). The reader
+will be responsible for checking the system's configuration and building a {Y2Network::Config}
+object, containing interfaces, configurations, routes, etc. On the other hand, the writer will be
+responsible for updating the system using that configuration object.
+
+As a developer, you rarely will need to access to readers/writers because `Yast::Lan` already offers
+an API to read and write the configuration. See the [Accessing the
+Configuration](#accessing-the-configuration) section for further details.
+
+### Sysconfig
+
+The sysconfig backend support is composed by these files:
 
     src/lib/y2network/sysconfig
     ├── config_reader.rb <- READER
     ├── config_writer.rb <- WRITER
-    ├── connection_config_reader_handlers
-    │   ├── eth.rb
-    │   ├── wlan.rb
-    │   └── ...
     ├── connection_config_reader.rb
+    ├── connection_config_readers
+    │   ├── eth.rb
+    │   └── wlan.rb
+    ├── connection_config_writer.rb
+    ├── connection_config_writers
+    │   ├── eth.rb
+    │   └── wlan.rb
     ├── dns_reader.rb
     ├── dns_writer.rb
     ├── interface_file.rb
     ├── interfaces_reader.rb
-    ├── interfaces_writer.rb
     └── routes_file.rb
 
-As you can see, there are many classes, but the relevant ones are just `ConfigReader` and `ConfigWriter`.
+{Y2Network::Sysconfig::ConfigReader} and {Y2Network::Sysconfig::ConfigWriter} are the reader and
+writer classes. Each of them cooperates with a set of ancillary classes in order to get the job
+done.
+
+{Y2Network::Sysconfig::DNSReader}, {Y2Network::Sysconfig::InterfacesReader} and
+{Y2Network::Sysconfig::ConnectionConfigReader} are involved in reading the configuration. The logic
+to read the configuration for a connection (e.g., `ifcfg-eth0`, `ifcfg-wlan0`, etc.) is implemented
+in a set of smaller classes (one for each time of connection) under
```

Comment:
maybe mention here InterfaceType class that is used for that smaller class as delimeter.